src/transformers/trainer.py

```diff
@@ -1182,6 +1314,30 @@ def create_scheduler(self, num_training_steps: int, optimizer: torch.optim.Optim
         Args:
             num_training_steps (int): The number of training steps to do.
         """
+        if optimizer is not None and isinstance(optimizer, LayerWiseDummyOptimizer):
```

Comment:
Based on the logic here - it looks like this should be inside `get_scheduler` rather than here

Reply:
good catch !

Reply:
Fixed !